### ISSUE TITLE
 nbconvert requires lxml_clean_html

### ISSUE THREAD

**Issue by sjkim2322** [elyra-ai/elyra#3228]
**Describe the issue**

If you run the notebook component on a runtime image that does not have lxml 5.2.0 or lower or lxml_clean_html installed in advance, the following error occurs.

```
        "lxml.html.clean module is now a separate project lxml_html_clean.\n"
        "Install lxml[html_clean] or lxml_html_clean directly." 
```
**To Reproduce**
1. Prepare a Python runtime image of a clean environment (in my case it is “[docker.io/python:3.10.12](https://hub.docker.com/r/python)”).
2. Run the elyra notebook component with the above image set as the runtime image.
3. The above error occurs while initializing the notebook environment.

**Cause I'm guessing**
- [The lxml version is not specified in the requirements of nbconvert==6.5.1](https://github.com/jupyter/nbconvert/blob/6.5.1/setup.py#L239-L257) used by elyra.
Therefore, when lxml is newly installed, the latest version is installed, and this version requires `lxml_clean_html`.
- [the changes in lxml 5.2.0 version.](https://github.com/lxml/lxml/commit/73778681f14359fe6d16644e69aaca276eba525a#diff-94784e90d1743b9d5dba655c4dcabfad62a6703b14b96ab144e9d314713371a7)

**Expected behavior**
- I can pre-install it on the runtime image I will use, but since all images require work, it would be better to add it to elyra's requirements.

**Deployment information**
Describe what you've deployed and how:
 - Elyra version: [3.16.0.dev0]
 - Installation source: [PyPI]
 - Deployment type: [Kubeflow [notebook server] ]
 - Operating system: [linux]

**Comment by lresende** [comment]
Should this be in Elyra? Or in nbconvert ?

**Comment by sjkim2322** [comment]
@lresende 
Thank you for answer.
I will also raise the issue on nbconvert.

However, I don't know if it is possible to modify the module dependency of a specific version that has already been deployed

In elyra, when the Jupyter notebook component is executed, it appears to download and install the requirements [here](https://github.com/elyra-ai/elyra/blob/main/etc/generic/requirements-elyra.txt). Since nbconvert is being installed here, I think it would be good if lxml_clean_html is also added here. 

or, If there is a spec that can pre-initialize the notebook component's cell before execution, it may be possible to use it.

**Comment by lresende** [comment]
As you can see, we already removed a lot of transient dependencies from the requirements file to avoid us having to keep syncing to new versions, etc... if they don't enable that in nbconvert itself, then we can continue this conversation.

**Comment by sjkim2322** [comment]
Yes, I left an issue on nbconvert. 
https://github.com/jupyter/nbconvert/issues/2148

**Comment by lresende** [comment]
@sjkim2322 based on your nbcovert issue, can't we update the version of nbconvert? 
does that still work with JupyterLab < 4?

**Comment by shalberd** [comment]
@lresende @sjkim2322 

I can confirm that we can just update the version of nbconvert without any issues regarding runtime behavior with the runtime image.

I can specifically confirm that nbconvert in a higher version works fine with Jupyterlab less than 4. See this code here from Red Hat Open Data Hub folks

https://github.com/opendatahub-io/notebooks/blob/main/runtimes/minimal/ubi9-python-3.9/utils/requirements-elyra.txt

**Comment by shalberd** [comment]
can be easily fixed by setting nbconvert to 7.1.0 https://github.com/jupyter/nbconvert/issues/2148
